newton/_src/geometry/broad_phase_sap.py

```
@@ -437,57 +462,57 @@ def __init__(
 
     def launch(
         self,
-        geom_lower: wp.array(dtype=wp.vec3, ndim=1),  # Lower bounds of geometry bounding boxes
-        geom_upper: wp.array(dtype=wp.vec3, ndim=1),  # Upper bounds of geometry bounding boxes
-        geom_cutoffs: wp.array(dtype=float, ndim=1),  # Cutoff distance per geometry box
-        geom_collision_group: wp.array(dtype=int, ndim=1),  # Collision group ID per box
-        geom_shape_world: wp.array(dtype=int, ndim=1),  # World index per box
-        geom_count: int,  # Number of active bounding boxes
+        shape_lower: wp.array(dtype=wp.vec3, ndim=1),  # Lower bounds of shape bounding boxes
+        shape_upper: wp.array(dtype=wp.vec3, ndim=1),  # Upper bounds of shape bounding boxes
+        shape_contact_margin: Optional[wp.array(dtype=float, ndim=1)],  # Optional per-shape contact margins
```

Ruff (0.14.5)

Undefined name `Optional`

(F821)